contracts/CollectiveGovernance.sol

```diff
@@ -89,8 +92,10 @@ contract CollectiveGovernance is Governance, VoteStrategy, ERC165 {
         Storage _governanceStorage
     ) {
         _voterClass = _class;
+        _storage = _governanceStorage;
+        uint256 _timeLockDelay = max(_storage.minimumVoteDuration(), Constant.TIMELOCK_MINIMUM_DELAY);
```

Comment:
what about a maximum delay?

Reply:
That is built into the TimeLock.    The TimeLock does not allow the lock duration to be 'out of bounds'.   Also once an event is scheduled it must be between the lockout period and the GRACE. So MAXIMUM + GRACE is the longest possible future schedule time for the 'nonce' in any TimeLock.